### PR TITLE
feat(ci): analyses snapshot skip pr on fail label

### PR DESCRIPTION
**Pull request by y3rsh**
## Overview

- ~~Keep making the analyses snapshot test workflow more flexible allowing a label on the PR to prevent failure PRs from opening. Adding the label `no-analyses-snapshot-pr` on your PR will cause the workflow to skip opening a PR for snapshot updates if the snapshot tests fail.~~
- Added a step to diff the snapshots in the PR target branch against the PR branch.  No action is taken based on the outcome. Purely informational.  Will this information be noticed and used as a reminder to synchronize with the target branch?

### Updates from feedback

- default `OPEN_PR_ON_FAILURE` to false 
- as is the case currently if the analyses snapshot test fails you will see a red X on that test and you can read the logs or download and view the HTML test report.
- if you want to open a PR into your branch that would heal the snapshots, add `gen-analyses-snapshot-pr` label and a PR will get opened
- create a specific step to handle PRs for the overnight scheduled test of edge.

### Risk

No risk or change to application code, CI only

### CODE REVIEW

**Comment by y3rsh** [comment]
### Proof
- <https://github.com/Opentrons/opentrons/actions/runs/10084931967/job/27884729239>
  - Has successful run of the new informational step.
- <https://github.com/Opentrons/opentrons/actions/runs/10085015669/job/27884972771?pr=15792#step:2:20>
  - Shows if you have the label on the PR the Flag is set correctly.

**Review by SyntaxColoring — APPROVED** [review]
Cool, thanks!

I have been thinking that this behavior should be the default. Like, we'd somehow get it so that if a snapshot test fails, you get *just a normal red X* in the CI section. And if you want a PR, you need to take some deliberate action, like commenting with a `/slashcommand`, tagging the PR with a magic tag, clicking a magic link, merging from a magic branch, or just doing it locally.

Because a recurring situation seems to be:

1. Analysis snapshot tests start failing on `edge` for whatever reason
2. 5 people branch their PRs off of `edge`
3. The snapshot tests fail in those 5 people's PRs
4. The system creates 5 snapshot PRs

(1)-(3) happen occasionally for normal tests and it's merely mildly annoying; (4) is the thing that's been uniquely noisy about snapshot tests.

**Review by DerekMaggio — APPROVED** [review]
oooh... fancy. 
I do agree with Max about how noisy the snapshots are.
I like the idea of a /command on the PR

**Comment by sfoster1** [comment]
Agree with @DerekMaggio and @SyntaxColoring 

Also can we get the opened PR (if it does exist) to link to the _pr_ for the branch that it's targeting, if possible? And ideally backlink to itself in the comment on that PR?

**Comment by y3rsh** [comment]
Addressed the comments.  Went with label as it is more visible than a comment and easier to handle for this binary circumstance.

File: analyses-snapshot-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[0a9ef592c8][Flex_S_v2_18_Illumina_DNA_Prep_48x].json [hunk 1]
```diff
@@ -1,17 +1,5 @@
...
-      "completedAt": "TIMESTAMP",
-      "createdAt": "TIMESTAMP",
-      "id": "UUID",
-      "key": "50c7ae73a4e3f7129874f39dfb514803",
-      "notes": [],
-      "params": {},
-      "result": {},
-      "startedAt": "TIMESTAMP",
-      "status": "succeeded"
-    },
```
Comment (SyntaxColoring) [inline analyses-snapshot-testing/tests/__snapshots__/analyses_snapshot_test/test_analysis_snapshot[0a9ef592c8][Flex_S_v2_18_Illumina_DNA_Prep_48x].json]: Reminder to undo this if it was just a change for testing.

File: .github/workflows/analyses-snapshot-test.yaml [hunk 2]
```diff
@@ -77,22 +99,22 @@ jobs:
...
-        if: always() && steps.handle_failure.outcome == 'success' && env.OPEN_PR_ON_FAILURE
+        if: always() && steps.handle_failure.outcome == 'success' && env.OPEN_PR_ON_FAILURE && github.event_name == 'pull_request'
         uses: peter-evans/create-pull-request@v6
         with:
-          commit-message: 'fix(analyses-snapshot-testing): snapshot failure capture'
-          title: 'fix(analyses-snapshot-testing): ${{ env.ANALYSIS_REF }} snapshot failure capture'
-          body: 'This PR is an automated snapshot update request. Please review the changes and merge if they are acceptable or find your bug and fix it.'
+          commit-message: 'fix(analyses-snapshot-testing): heal analyses snapshots'
+          title: 'fix(analyses-snapshot-testing): heal ${{ env.ANALYSIS_REF }} snapshots'
+          body: 'This PR was requested on the PR https://github.com/${{ github.repository }}/pull/${{ github.event.pull_request.number }}'
```
Comment (SyntaxColoring) [inline .github/workflows/analyses-snapshot-test.yaml:112]: Very nice.

**Review by SyntaxColoring — APPROVED** [review]
Thank you!

**Pull request merged**